onnx/shape_inference/implementation.cc

```diff
@@ -426,7 +427,9 @@ static void InferShapesImpl(
         materializeSymbolicShape(inferredType, symbolTable);
         // Now we can merge pre-existing and inferred info
         mergeShapesAndTypes(*inferredType, existingType);
-
+        DataPropagationContextImpl dataPropagationCtx(
+            n, valueTypesByName, inputDataByName, generatedShapeDataByName, &graphInferenceContext);
+        schema->GetDataPropagationFunction()(dataPropagationCtx);
```

Comment:
every op will not have a data propagator defined... add a check something like
schema->has_data_propagation_function() before creating dataPropagationCtx and calling schema->GetDataPropagationFunction()(dataPropagationCtx);

Reply:
Good catch. Added in both Python and C++.